kubernetes/data_source_kubernetes_ingress.go

```diff
@@ -0,0 +1,27 @@
+package kubernetes
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func dataSourceKubernetesIngress() *schema.Resource {
+	s := resourceKubernetesIngress().Schema
+	s["metadata"] = namespacedMetadataSchema("ingress", false)
+	s["spec"].Computed = true
+	s["spec"].Required = false
+	return &schema.Resource{
+		Read:   dataSourceKubernetesIngressRead,
+		Schema: s,
```

Comment:
In other data sources, we have a convention to write out the schema explicitly. Here's an example: https://github.com/terraform-providers/terraform-provider-kubernetes/blob/master/kubernetes/data_source_kubernetes_service_account.go#L14 It would be great if you could write out this schema similarly. Thanks!

Reply:
Okay, I was adressing the request / recommendation of @alexsomesan 

> alexsomesan commented on 15 Oct 2019
> 
> Thanks for fixing the tests. They do pass now.
> 
> One more question: is there any reason for not sharing the schema definition with the Ingress resource?

So I reused the schema definition from the Ingress resource, also to adress upcoming extensions of the ingress.

Reply:
@dak1n1 @alexsomesan What is your prefered solution? 
Sharing the schema definition with the ingress resource? or write the schema explicitly?

Your feedback is warmly welcome.

Reply:
Sorry, it looks like I gave you conflicting advice! I know Alex is really busy lately, so he might not be able to respond for a while. I wonder if @jrhouston might be able to give us his opinion instead.

My opinion is to structure this similarly to the other data sources, mostly for consistency and because we know that structure works. I think sharing a schema probably would obscure more issues than it would solve, because the data source will always need different settings than the resource. For example, resource schema attributes each have `optional: true`, and the correlating data sources have `computed: true` instead. I know you were able to overwrite at least one of those, but I'm not sure what impact the rest of them will have.

Reply:
I have now followed the implementation style of the other data_source_kubernetes_* objects and written the schema explicitly. 

For me unit testing and also testing in real kubernetes clusters is working as expected. 

@dak1n1 @jrhouston @alexsomesan

Reply:
I think for consistency it is good to write the schema out explicitly. You can save some space by pulling the documentation from the SwaggerDoc though, there is an example [here](https://github.com/terraform-providers/terraform-provider-kubernetes/blob/master/kubernetes/resource_kubernetes_validating_webhook_configuration.go#L18).

Reply:
@jrhouston Do you see the documentation with SwaggerDoc as required for this PR?